include/nbl/ext/TextRendering/TextRendering.h

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_EXT_TEXT_RENDERING_H_INCLUDED_
+#define _NBL_EXT_TEXT_RENDERING_H_INCLUDED_
+
+#include "nabla.h"
+
+#include "nbl/video/alloc/SubAllocatedDataBuffer.h"
+#include "nbl/video/utilities/CPropertyPool.h"
+#include <msdfgen/msdfgen.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_OUTLINE_H
+
+using namespace nbl;
+using namespace nbl::core;
+using namespace nbl::asset;
+using namespace nbl::video;
+
+namespace nbl
+{
+namespace ext
+{
+namespace TextRendering
+{
+
+constexpr uint32_t asciiAtlasCharacterCount = (int('~') - int(' ')) + 1;
+
+struct SPixelCoord
+{
+	uint16_t x, y;
+};
+
+class NBL_API FontAtlas
+{
+public:
+	FontAtlas(IGPUQueue* queue, ILogicalDevice* device, const std::string& fontFilename, uint32_t glyphWidth, uint32_t glyphHeight, uint32_t charsPerRow);
+private:
+	FT_Library library;
+	FT_Face face;
+
+	std::array<SPixelCoord, asciiAtlasCharacterCount> characterAtlasPosition;
```

Comment:
Can it be defined mathematically? I think not because of msdfgen, I think we also need this as an `IGPUBufferView` later?

Reply:
no because not every character gets a glyph

Reply:
Based on the pseudo-code fragment shader we need:
1. `IGPUBufferView/usamplerBuffer` with format `rg16_uint` for glyph pixel sizes.
2. IGPUImage for atlasImage which you already have,

Reply:
Updated the vertex shader codelisting in the issue btw